src/qtwebengine/dataprovider.cpp

```diff
@@ -26,13 +72,23 @@ void DataProvider_QWebEngine::requestStarted( QWebEngineUrlRequestJob *request )
         return;
     }
 
-    QString mimetype;
+    QByteArray mimetype;
 
-    // Specify the encoding in case the page content is not UTF-8
+    // We must specify the proper MIME type for the page to display correctly.
+    // The HTML and XML files correspond to "text/html";
+    // for other types "application/octet-stream" is sufficient.
+    // In addition, for "text/html", a "meta" tag is added specifying the text encoding.
+    // This is the easiest and most stable way to set the encoding.
     if ( htmlfile )
-        mimetype = QString( "text/html; charset=%1" ) .arg( ::mainWindow->chmFile()->currentEncoding() );
```

Comment:
Are you sure this works?

Reply:
I tested the application in the following environment:

- CPU : Intel(R) Pentium(R) Dual CPU E2180 @ 2.00GHz
- RAM : 3GB
- OS : Linux Mint 18.1 Serena with the kernel "Linux 4.4.0-53-generic i686 GNU/Linux"
- DE : MATE Desktop Environment 1.16.2
- Qt : local build 5.15.2

And this case I checked especially carefully.

Reply:
Have you checked that the new code indeed changes anything within the detection?